cdap-api/src/main/java/co/cask/cdap/api/service/http/HttpContentConsumer.java

```diff
@@ -91,6 +93,7 @@
    * @param responder a {@link HttpServiceResponder} for sending response
    * @throws Exception if there is any error
    */
+  @TransactionPolicy(TransactionControl.IMPLICIT)
```

Comment:
Do you want to add `@TransactionPolicy(TransactionControl.EXPLICIT)` to the `onReceived` method above?